piqe_ocp_lib/api/resources/ocp_node_metrics.py

```diff
@@ -0,0 +1,50 @@
+from piqe_ocp_lib.api.resources.ocp_base import OcpBase
+from kubernetes.client.rest import ApiException
+import logging
+from piqe_ocp_lib import __loggername__
+
+logger = logging.getLogger(__loggername__)
+
+
+class OcpNodeMetrics(OcpBase):
+    """
+    OcpNodeMetrics Class extends OcpBase and encapsulates all methods
+    related to managing Openshift nodes Metrics.
+    :param kube_config_file: A kubernetes config file.
+    :return: None
+    """
+
+    def __init__(self, kube_config_file=None):
+        super(OcpNodeMetrics, self).__init__(kube_config_file=kube_config_file)
+        self.api_version = 'metrics.k8s.io/v1beta1'
+        self.kind = 'NodeMetrics'
+        self.ocp_nodes = self.dyn_client.resources.get(api_version=self.api_version, kind=self.kind)
+
+    def get_a_node_metrics(self, node_name):
+        """
+        Method returns a node metrics by name for specific node
+
+        :param node_name: The name of the node.
+        :return: node_metric_response (ResourceInstance) on success OR None on failure.
+        """
+        node_metric_response = None
+        try:
+            node_metric_response = self.ocp_nodes.get(name=node_name)
+        except ApiException as e:
+            logger.error("Exception encountered while getting node metrics by name: %s\n", e)
+        logger.debug("Node metric response : %s", node_metric_response)
+        return node_metric_response
+
+    def get_node_metrics(self):
```

Comment:
I agree with Adam's comment on naming convention. I find that better names would be 'get_node_metrics' and 'get_nodes_metrics' for single and all nodes respectively.

Reply:
I used name to be in line with ocp_nodes class to avoid this confusion. Here are from ocp_nodes.py class : 

 def get_a_node(self, node_name):
        """
        Method returns a node object by name

        :param node_name: The name of the node.
        :return: V1Node on success. None on failure.
        """
    def get_all_nodes(self, label_selector=None):
        """
        Method that returns a list of node objects
        :param label_selector: Used to return a a list of nodes based on the provided label(s)
        :return: V1NodeList on success. None on failure.
        """

I can update the method name.

Reply:
Updated the method names.